src/plugins/index_pattern_field_editor/public/service/field_format_editors/field_format_editors.ts

```diff
@@ -6,16 +6,16 @@
  * Side Public License, v 1.
  */
 
-import { DefaultFormatEditor } from '../../components/field_format_editor';
+import { FieldFormatEditorFactory } from '../../components/field_format_editor';
 
 export class FieldFormatEditors {
-  private editors: Array<typeof DefaultFormatEditor> = [];
+  private editors: FieldFormatEditorFactory[] = [];
```

Comment:
Now instead of registering a component, we require to register a factory that creates the component. This allows to load the component async